sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample7_ConsumeEvents.cs

```diff
@@ -0,0 +1,132 @@
+            // of the consumer in an application, such as "Telemetry" or "OrderProcessing".  When an Event Hub is created, a default
+            // consumer group is created with it, called "$Default."
+            //
+            // Each consumer has a unique view of the events in the partition, meaning that events are available to all consumers
+            // and are not removed from the partition when a consumer reads them.  This allows for different consumers to read and
+            // process events from the partition at different speeds and beginning with different events without interfering with
+            // one another.
+            //
+            // When events are published, they will continue to exist in the partition and be available for consuming until they
+            // reach an age where they are older than the retention period.
+            // (see: https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-faq#what-is-the-maximum-retention-period-for-events)
+            //
+            // Because events are nor removed from the partition when consuming, a consumer must specify where in the partition it
+            // would like to begin reading events.  For example, this may be starting from the very beginning of the stream, at an
+            // offset from the beginning, the next event available after a specific point in time, at a specific event.
+
+            // We will start by creating a client each using its default set of options.
+
+            await using (var client = new EventHubClient(connectionString, eventHubName))
+            {
+                // With our client, we can now inspect the partitions and find the identifier
+                // of the first.
+
+                string firstPartition = (await client.GetPartitionIdsAsync()).First();
+
+                // In this example, we will create our consumer for the first partition in the Event Hub, using the default consumer group
+                // that is created with an Event Hub.  Our consumer will begin watching the partition at the very end, reading only new events
+                // that we will publish for it.
+
+                await using (EventHubConsumer consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroup, firstPartition, EventPosition.Latest))
+                await using (EventHubProducer producer = client.CreateProducer(new EventHubProducerOptions { PartitionId = firstPartition }))
+                {
+                    // Because our consumer is reading from the latest position, it won't see events that have previously
+                    // been published.  Before we can publish the events, we will need to ask the consumer to perform an operation,
+                    // because it opens its connection only when it needs to.  The first receive that we ask of it will not see
+                    // any events, but will allow the consumer to start watching the partition.
```

Comment:
You might want to explain that the receive below will return immediately.

Reply:
Good  call.   Added to both 7 and 8.